config/src/network_id.rs

```diff
@@ -2,19 +2,43 @@
 // SPDX-License-Identifier: Apache-2.0
 use crate::config::RoleType;
 use libra_types::PeerId;
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Serialize, Serializer};
 use std::fmt;
 
 /// A grouping of common information between all networking code for logging.
 /// This should greatly reduce the groupings between these given everywhere, and will allow
 /// for logging accordingly.
-#[derive(Clone, Eq, PartialEq, Serialize)]
+#[derive(Clone, Eq, PartialEq)]
 pub struct NetworkContext {
     network_id: NetworkId,
     role: RoleType,
     peer_id: PeerId,
 }
 
+/// This custom serializer is to make sure that there is a consistent output type of
```

Comment:
might be easier to use a serde field attribute serialize_with and just point to how you want the network_id to look. https://serde.rs/field-attrs.html

Reply:
Yeah, that probably was easier :P